### DISCUSSION TITLE
Modify text from different thread

### DISCUSSION THREAD

**Discussion by chnbr** [danielsaidi/RichTextKit#219] [Q&A]
I have an app that has dictation support. I want to 'inject' text from my dictation engine, which is running in a completely different thread.

Currently it looks like context must be declared in its displaying View to work properly. 
How do I get access to context in another thread ? 
Is context multi thread safe ?

Thanks

**Comment by danielsaidi** [comment]
Can you perform the dictation result update on the main actor?

**Comment by chnbr** [comment]
I have a solution. Put the stuff into a model and it works from both threads. Thanks

**Comment by chnbr** [comment]
Another (related) question. I need to get notified on clicks and movements of the cursor as well as keyboard input events because I need to stop modifications of the text from the other thread.

How do I accomplish this with RichTextEditorKit ?  
Observing isEditing property will not be sufficient, because one could move the cursor while already in editing mode.
Observing raw text is also not an option because that is going to be changed by the other thread asynchronously.

**Reply by danielsaidi** [reply]
Hmmm I'm not sure :/
